tracer/src/Datadog.Trace/SpanContextInjector.cs

```diff
@@ -0,0 +1,37 @@
+// <copyright file="SpanContextInjector.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Datadog.Trace.DataStreamsMonitoring;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Propagators;
+using Datadog.Trace.SourceGenerators;
+using Datadog.Trace.Telemetry;
+using Datadog.Trace.Telemetry.Metrics;
+
+#nullable enable
+
+namespace Datadog.Trace
+{
+    /// <inheritdoc />
+    public class SpanContextInjector : ISpanContextInjector
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<SpanContextInjector>();
+
+        /// <inheritdoc />
+        [PublicApi]
+        public void Inject<TCarrier>(TCarrier carrier, Action<TCarrier, string, string> setter, ISpanContext context)
+        {
+            TelemetryFactory.Metrics.Record(PublicApiUsage.SpanContextInjector_Inject);
+
+            if (context is SpanContext spanContext)
+            {
+                SpanContextPropagator.Instance.Inject(spanContext, carrier, setter);
+            }
```

Comment:
I think we need to add support for data streams monitoring in here, but I haven't looked into _how_ 😬

Reply:
I don't think we should. I don't believe custom streams would be supported by DSM.